### PR TITLE
err assigned and not used

### PR DESCRIPTION
**Pull request by astaxie**
1. Miss a lot of err check in test case
2. return value and local variable mixed

please review

### CODE REVIEW

File: expression/builtin_math.go [hunk 1]
```diff
@@ -810,16 +810,14 @@ func (b *builtinAtanSig) eval(row []types.Datum) (d types.Datum, err error) {
 		x, err := args[1].ToFloat64(sc)
 		if err != nil {
 			d.SetFloat64(math.Atan(y))
-			// FIXME: Trigger a warning perhaps.
```
Comment (coocood) [inline expression/builtin_math.go]: Is this removed by accident?

**Comment by shenli** [comment]
LGTM

File: expression/builtin_math.go (hunk 1 above)
Comment (astaxie) [inline expression/builtin_math.go]: No, I think this trigger warning is the `err assigned and not used`

Comment (coocood) [inline expression/builtin_math.go]: But the error still not used.

Comment (coocood) [inline expression/builtin_math.go]: Well, I mean not handled.
It is not assigned anymore.

**Comment by astaxie** [comment]
I found the travis test failed in these two case:

```
builtin_string_test.go:1409:
    c.Assert(err, IsNil)
... value *errors.Err = &errors.Err{message:"", cause:(*errors.Err)(0xc4200da0f0), previous:(*errors.Err)(0xc4200da0f0), file:"github.com/pingcap/tidb/expression/_test/_obj_test/builtin_string.go", line:2118} ("cannot convert [{5 0 0 0 0 [49 48] <nil>}](type []types.Datum) to int64")
```

```
builtin_time_test.go:1190:
    c.Assert(err, IsNil)
... value *errors.Err = &errors.Err{message:"", cause:(*errors.Err)(0xc42005cf00), previous:(*errors.Err)(0xc4200da5a0), file:"github.com/pingcap/tidb/expression/_test/_obj_test/builtin_time.go", line:2400} ("invalid time format")
```

Does the code has issues or this modify cause the issue?

If the `f.eval(nil)` expect to error, do we need to change the test case with errors?

**Comment by hanfei1991** [comment]
@astaxie The travis CI will run your test with the newest master. Merge master maybe you can find the error.

**Comment by coocood** [comment]
@astaxie 
The first failed test case has issues,  the second failed test case expects error.

**Comment by astaxie** [comment]
> The first failed test case has issues, 

So the test case is right. Then it's your guys problem. 🙈 

> the second failed test case expects error.

If it expect error then the need to change the test case with expect errors

**Comment by coocood** [comment]
@astaxie 
Yes, it's not your fault that the CI failed.😂

**Comment by coocood** [comment]
LGTM

@astaxie 
Thank you!

File: expression/builtin_math.go (hunk 1 above)
Comment (astaxie) [inline expression/builtin_math.go]: But this comments `Trigger a warning perhaps` not make sense here. For your case here you just use the err as a judging condition, but you never use it's value as it's a local variable from line 810-815

**Comment by astaxie** [comment]
🍻

**Pull request merged**